nixos/modules/services/networking/mptcp/mptcp_up_raw

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+# A script for setting up routing tables for MPTCP
+
+# can check with (ins)[root@client:~]# journalctl -b -u NetworkManager-dispatcher.service
+# Copy this script into /etc/network/if-up.d/
+
+set -ex
+
+STATUS=$2
+
+# TODO remove
+echo $PATH
+env > /tmp/if_up_env
+
+RT_TABLE=/etc/iproute2/rt_tables
+
+if [ "$DEVICE_IFACE" = "lo" ]; then
+
+	logger -p user.debug -t mptcp "if localhost or $MODE then abort "
+	exit 0
+fi
+
+if [ -z "$DEVICE_IFACE" ]; then
+	logger -p user.warn -t mptcp "invalid $DEVICE_IFACE"
+	exit 0
+fi
+
+
+if [ `grep "$DEVICE_IFACE" "$RT_TABLE" | wc -l` -eq 0 ]; then
+	logger -p user.info -t mptcp "Adding new routing table $DEVICE_IFACE"
+	NUM=$(wc -l < "$RT_TABLE")
+	echo "$NUM  $DEVICE_IFACE" >> "$RT_TABLE"
```

Comment:
Why do we need this to be created at runtime rather then controlling the whole `etc/iproute2/rt_tables` file?

Reply:
By default `/etc/iproute2/rt_tables` does not exists, which would make:

```
NUM=$(wc -l < /etc/iproute2/rt_tables)
zsh: no such file or directory: /etc/iproute2/rt_tables
```

fail

Reply:
Is it needed at all to maintain `rt_table` for this to work?

Reply:
might be fixed by `networking.iproute2.enable = true;` so.